datarobot_batch_scoring/writer.py

```diff
@@ -410,53 +416,108 @@ def format_result_data(self, result, batch, pred_name):
                            key=operator.itemgetter('row_id'))]
             out_fields = ['row_id', pred_name if pred_name else '']
         else:
-            ValueError('task "{}" not supported'
-                       ''.format(result['task']))
-        return (out_fields, pred)
+            raise ValueError('task "{}" not supported'
+                             ''.format(result['task']))
 
-    def sigterm_handler(self, _signo, _stack_frame):
```

Comment:
I think removing signal handling is a mistake. Without it you cannot perform a cleanup if anything besides keyboardinterupt/SIGINT is received by the master.
If you SIGKILL the master, the workers keep going like nothing happened because the master dies immediately without any cleanup.
If you SIGKILL/SIGINT a workers, it just dies without triggering a program shutdown. The program only exits much later.
Every process should have a way to catch signals and trigger a fast, but safe, shutdown. 

Probably workers should be able to forward the signal to the parent.

If they SIGKILL they are on their own :)